src/App.tsx

```diff
@@ -1,61 +1,159 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { User, Todo } from './types/types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+const initialTodos = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [errorUser, setErrorUser] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setTitle(value);
+    setErrorTitle(false);
+  };
+
+  const resetFields = () => {
+    setTitle('');
+
+    setErrorUser(false);
+    setSelectedUserId(0);
+  };
+
+  const handleUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setErrorUser(false);
+    setSelectedUserId(+value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUserId) {
+      setErrorUser(true);
+    }
+
+    setTitle(title.replace(/[^A-Za-z\s\d\u0400-\u04FF]/g, ''));
```

Comment:
Move the `setTitle` call inside the `if (title.trim() && selectedUserId)` block to avoid unnecessary updates.

Reply:
or use `return` from function after settign error